sphinx/ext/inheritance_diagram.py

```diff
@@ -219,21 +362,47 @@ def get_all_class_names(self):
         'height': 0.25,
         'fontname': '"Vera Sans, DejaVu Sans, Liberation Sans, '
                     'Arial, Helvetica, sans"',
-        'style': '"setlinewidth(0.5)"',
+        'style': [ 'setlinewidth(0.5)' ],
     }
     default_edge_attrs = {
         'arrowsize': 0.5,
-        'style': '"setlinewidth(0.5)"',
+        'style': [ 'setlinewidth(0.5)' ],
+    }
+    builtin_node_attrs = {
+        'style': [ 'dashed' ],
+    }
+    express_edge_attrs = {
+        'style': [ 'dotted' ],
     }
 
-    def _format_node_attrs(self, attrs):
-        return ','.join(['%s=%s' % x for x in attrs.items()])
-
-    def _format_graph_attrs(self, attrs):
-        return ''.join(['%s=%s;\n' % x for x in attrs.items()])
+    def _attr_merge(self, *args):
+        merged = {}
+        for attribute in args:
+            for k,v in attribute.items():
+                if  (   k in merged
+                    and all(    (   isinstance(i, collections.abc.Iterable)
+                                and not isinstance(i, str)
```

Comment:
This should be s/str/string_types/ to be 2.x and 3.x compatible (there's another instance of this check).